### ISSUE TITLE
After upgrading to react 18, tooltip not disappearing

### ISSUE THREAD

**Issue by zsofiath** [wwayne/react-tooltip#758]
I created a react app with CRA, installed react-tooltip, and used this code snippet:

![image](https://user-images.githubusercontent.com/17666549/162138545-9bee4ed6-2d3a-4317-99e9-535dd97dc168.png)

After mouseleave event, the tooltip remains visible.

I've found the snippet here:
https://codesandbox.io/s/heuristic-curran-bddeu?fontsize=14&hidenavigation=1&theme=dark

**Comment by amirho1** [comment]
same here it doesn't disappear in fresh create-react-app

**Comment by amirho1** [comment]
what I understand about this issue is; it's only in development and it doesn't have any problems in production build test it I red some were but I didn't test it @Suzan-Dev

**Comment by Suzan-Dev** [comment]
@amirho1 Yeah, same here. I read about that here #763 but still choose to downgrade to v17. React 18 is here but packages are not up to date. 😞😞

**Comment by vova42matters** [comment]
+1

**Comment by svmszcck** [comment]
As a quick fix you can do this until the package is updated:

```
const [tooltip, showTooltip] = useState(true);

<>
 {tooltip && <ReactTooltip effect="solid" />}
 <p
   data-tip="hello world"
   onMouseEnter={() => showTooltip(true)}
   onMouseLeave={() => {
     showTooltip(false);
     setTimeout(() => showTooltip(true), 0);
   }}
 />
</>
```

**Comment by vova42matters** [comment]
In my case had an issue with z-index (there was an element which was overflowing tooltip), just double-check z-indexes for elements

**Comment by briva** [comment]
same problem here

**Comment by amirho1** [comment]
@briva and others who came here, I think it doesn't work on the development, Please check it on production, it seems; it's working correctly

**Comment by voldo5** [comment]
I have the same problem. But it works fine after removing <React.StrictMode> from root.

**Comment by angelangelov88** [comment]
I have the same issue. Thanks @svmszcck for the tip

**Comment by danielbarion** [comment]
Hi, @zsofiath  can you try the latest version of the package and let us know, please?

**Comment by angelangelov88** [comment]
Hi @zsofiath,

I was having this same issue previously and had to use react-tooltip-rc to make it work properly on React 18. Have just tried again with react-tooltip and it all works great now. 

Thanks a lot!

**Comment by danielbarion** [comment]
Hi guys, regarding the @angelangelov88 comment, I'll close that issue.

Please feel free to open a new one if needed. Thanks!
